lib/flagsmith.rb

```diff
@@ -1,121 +1,188 @@
 # frozen_string_literal: true
 
+require 'semantic'
 require 'faraday'
 require 'faraday_middleware'
+require 'securerandom'
+require 'pry'
+
+# require_relative 'flagsmiths/analytics'
+# require_relative 'flagsmiths/errors'
+# require_relative 'flagsmiths/models/flag'
+# require_relative 'flagsmiths/models/flags/collection'
+# require_relative 'flagsmiths/pooling_manager'
+# require_relative 'flagsmiths/api_client'
+# require_relative 'flagsmiths/config'
+# require_relative 'flagsmiths/helpers'
+
+Dir.glob('./lib/flagsmiths/client/**/*.rb').sort.each { |file| require file }
+Dir.glob('./lib/flagsmiths/engine/**/*.rb').sort.each { |file| require file }
+
+require_relative 'flagsmith_engine'
 
 # Ruby client for flagsmith.com
 class Flagsmith
-  attr_reader :flagsmith_api
-
-  def initialize(opts = {})
-    @opts = determine_opts(opts)
-
-    @flagsmith_api = Faraday.new(url: @opts[:url]) do |faraday|
-      faraday.headers['Accept'] = 'application/json'
-      faraday.headers['Content-Type'] = 'application/json'
-      faraday.headers['x-environment-key'] = @opts[:api_key]
-      faraday.response :json
-      # TODO: add timeout adjustment here
-      faraday.adapter Faraday.default_adapter
-    end
+  extend Forwardable
+  include FlagsmithEngine
+  include Flagsmiths::Helpers
+  # A Flagsmith client.
+  #
+  # Provides an interface for interacting with the Flagsmith http API.
+  # Basic Usage::
+  #
+  # flagsmith = Flagsmith.new(environment_key: '<your API key>')
+  # environment_flags = flagsmith.get_environment_flags
+  # feature_enabled = environment_flags.is_feature_enabled('foo')
+  # identity_flags = flagsmith.get_identity_flags('identifier', 'foo': 'bar')
+  # feature_enabled_for_identity = identity_flags.is_feature_enabled('foo')
+
+  # Available Configs.
+  #
+  # :environment_key, :api_url, :custom_headers, :request_timeout_seconds, :enable_local_evaluation,
+  # :environment_refresh_interval_seconds, :retries, :enable_analytics, :default_flag_handler
+  # You can see full description in the Flagsmiths::Config
+  #
+
+  attr_reader :config, :environment
+
+  delegate Flagsmiths::Config::OPTIONS => :@config
```

Comment:
That's for sure. This is because we have a main `Flagsmith` class that we can instantiate. We must change it to <`Flagsmith`> and create a new main class, such as `Flagsmith::SDK` / `Flagsmith::Client`. Let's discuss it tomorrow.